### ISSUE TITLE
any plans

### ISSUE THREAD

**Issue by alinelena** [rjfarmer/gfort2py#22]
Are there any plans for adding the missing bits?

and make it a proper modern fortran to py

**Comment by rjfarmer** [comment]
Plans? Yes

Time needed to implement said plans? No :)

**Comment by alinelena** [comment]
I wonder if anyone at fortran-lang is intereste @certik what do you think?

**Comment by certik** [comment]
Yes, I am interested and we definitely want this functionality. The question I have is if we want to invest into all these separate tools such as f2py, gfort2py, FORD, etc., or rather consolidate into one framework and just worry about a backend. 

I plan to have backend for exactly this in LFortran, see: https://gitlab.com/lfortran/lfortran/-/issues/133. LFortran will be able to parse gfortran module files: https://gitlab.com/lfortran/lfortran/-/issues/52, so it would effectively be able to do what `gfort2py` could. The reason I want to implement this as part of LFortran is that since LFortran will become a full Fortran compiler, I know that it will have all the bits and pieces to support all of Fortran correctly. While f2py is having its own parser (huge task, and never quite finished), gfort2py is better: it is essentially a backend to gfortran, so gfortran is a full compiler, it understands the code correctly, and gfort2py just provides the wrappers. But still it becomes quite a lot of work if you want to support all Fortran constructs, such as derived types and classes, etc. Since I have to do most of this work anyway for LFortran, I plan to implement the rest there.

The idea is that LFortran can either parse your Fortran source code (as a compiler) or extract the information from gfortran compiled mod files, and returns an ASR (Abstract Semantic Representation) to you. You then just write ASR -> Python wrappers.

**Comment by alinelena** [comment]
Personally I want the functionality where is done is relatively non important for me as long as it open source. f2py and other tools I have seen are a no go for me since modern means 1991 for them.
my  problem is my compiler programming is non existent so all I can do is test and break things in this. maybe we shall open a general discussion on fortran-lang?

**Comment by certik** [comment]
Yes, you can discuss this at fortran-lang also.
